mllib/src/main/scala/org/apache/spark/mllib/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -107,8 +107,11 @@ private[tree] object DecisionTreeMetadata extends Logging {
       numTrees: Int,
       featureSubsetStrategy: String): DecisionTreeMetadata = {
 
-    val numFeatures = input.take(1)(0).features.size
+    require(!input.isEmpty, s"DecisionTree requires size of input RDD > 0, " +
```

Comment:
Good point. I also think this line should be placed before "val numExamples = input.count()" thus there will be a fast fail.